### ISSUE TITLE
spaces in front of unique output?

### ISSUE THREAD

**Issue by janschulz** [IRkernel/repr#22]
When I use `unique(full_df$year)` or `names(full_df)`, I get ugly space in front of the list:

![2015-12-15_105553](https://cloud.githubusercontent.com/assets/890156/11807503/72ad1ef4-a31a-11e5-952f-2c92f3117883.png)

In the notebook, this is inserted as:
```
<div class="output_subarea output_html rendered_html output_result"><ol class="list-inline">
	<li>28418</li>
	<li>23651</li>
	<li>22405</li>
	<li>145671</li>
        <li>[...]</li>
</ol>
</div>
```

I've never seen such a "space" in python.

Also:

Rstudio:
```
> unique(c(1,2,3,4,5,1,2,3,4,5))
[1] 1 2 3 4 5
```
notebook:
```
In [2]: unique(c(1,2,3,4,5,1,2,3,4,5))
Out[2]:      1 2 3 4 5
```

-> the "[1]" in front of the output is missing

```
print("whatever")
```
Also prints a `"[1]"` in front of the output

**Comment by JanSchulz** [comment]
@takluyver commented 2 hours ago:

I think the [1] index numbers on the left only really make sense in a fixed width environment where the process producing the output knows how it will be layed out. That's not currently the case with the HTML representation of vectors, because the numbers can be reflowed according to the browser window size. Maybe we can come up with some alternative way of conveying similar information, though.

**Comment by JanSchulz** [comment]
@flying-sheep commented 11 minutes ago:

jup, that’s also the reason why i chose to use the enumitem package. there you can configure if it’s important enough to see the index or not.

**Comment by flying-sheep** [comment]
yeah, as said: i chose to represent vectors as `ol.list-inline` in HTML.

the alternatives would be non-inline lists (huge) or basically falling back to fixed-width display.

*samentically*, more information is preserved than in the plain text display (each vector entry is a list item) but it’s unclear how to display this information best.

i think the ideal way (reflowing output and automatically changing numbers in front of each line powered by JS) is a bit heavy, no?

**Comment by takluyver** [comment]
Maybe there could be little markers inline, say after every 100th element, a little superscript `[400]` in a different colour?

**Comment by JanSchulz** [comment]
My vote would go to the option, that the basic data structures (lines, vectors) should result in a printed output, not html... :-)

**Comment by flying-sheep** [comment]
@takluyver great idea!

we have the bootstrap classes

* .mark
* .text-primary
* .text-success
* .text-info
* .text-warning
* .text-danger

**Comment by JanSchulz** [comment]
Not sure what should happen here, but this also doesn't look constant

```
In [10]: print("text")
[1] "text"

In [11]: "text"
Out[11]: "text"
```

vs rstudio:

```
> print("text")
[1] "text"
> "text"
[1] "text"
```

**Comment by flying-sheep** [comment]
i don’t get it. `print()` prints stuff to stdout and the literal evaluates to itself.

what’s the problem here?

**Comment by JanSchulz** [comment]
I find it strange that the r console suggests that `"text"` is a vector (prefix `[1]`) but not the notebook.

**Comment by flying-sheep** [comment]
there is nothing in R that isn’t a vector except from NULL

**Comment by JanSchulz** [comment]
There is also this:

```
1:10
c(a=1,b=2)
list(a=1,b=2)
sapply(mtcars, class)
```

![2016-03-16_103550](https://cloud.githubusercontent.com/assets/890156/13807720/e083cb68-eb62-11e5-8111-906977f158e2.png)

-> should all be left aligned.

**Comment by flying-sheep** [comment]
the first one is fixed in jupyter/notebook#1220

the named vector repr could be different, yes. maybe like a table with one header row and one row?

**Comment by JanSchulz** [comment]
> your real issue is IRkernel/repr#22, but you still propose not doing rich output?

Yes, mainly because I don't see a benefit of having richt display for such primitives: they take more space and are not c&p friendly (tabs, etc). (It's even better with the python approach to output eval-able representations, but we had this already... :-))

For tables and other higher level stuff in most cases there is a clear benefit of rich output. Especially for things like regression tables where you want to compare stuff between rows and columns.
